OrbitCore/SymbolHelper.h

```diff
@@ -0,0 +1,45 @@
+#ifndef SYMBOL_HELPER_H_
+#define SYMBOL_HELPER_H_
+#include <string>
+#include <vector>
+
+#include "OrbitModule.h"
+#include "Pdb.h"
+
+class SymbolHelper {
+ public:
+  // TODO check function ordering
+  SymbolHelper();
+  SymbolHelper(std::vector<std::string> collector_symbol_directories,
+               std::vector<std::string> symbols_file_directories)
+      : collector_symbol_directories_(collector_symbol_directories),
+        symbols_file_directories_(symbols_file_directories){};
+
+  bool LoadSymbolsIncludedInBinary(std::shared_ptr<Module> module) const;
+  bool LoadSymbolsCollector(std::shared_ptr<Module> module) const;
+  bool LoadSymbolsUsingSymbolsFile(std::shared_ptr<Module> module) const;
+  void LoadSymbolsFromDebugInfo(std::shared_ptr<Module> module,
+                                const ModuleDebugInfo& module_info) const;
+  void FillDebugInfoFromModule(std::shared_ptr<Module> module,
+                               ModuleDebugInfo& module_info) const;
+
+  std::vector<std::string> GetCollectorSymbolDirectories() const {
+    return collector_symbol_directories_;
+  }
+  std::vector<std::string> GetSymbolsFileDirectories() const {
+    return symbols_file_directories_;
+  }
+
+  void SetCollectorSymbolDirectories(std::vector<std::string> directories) {
+    collector_symbol_directories_ = directories;
+  }
+  void SetSymbolsFileDirectories(std::vector<std::string> directories) {
+    symbols_file_directories_ = directories;
+  }
+
+ private:
+  std::vector<std::string> collector_symbol_directories_;
+  std::vector<std::string> symbols_file_directories_;
```

Comment:
which probably means you can make these const

Reply:
yes, the first one. The second vector is read from file.

Reply:
done with Hennings help